### PR TITLE
[VL] Upgrade Arrow version to 11.0.0

### PR DESCRIPTION
**Pull request by zhztheplayer**
Incorporates #946 and its fix #945.

At this time we have to maintain our own forked codes of Arrow since we have a QAT patch which is not yet upstreamed.

### CODE REVIEW

**Comment by FelixYBW** [comment]
Can we use upstream arrow, then apply patch to get it work?

**Comment by zhztheplayer** [comment]
> Can we use upstream arrow, then apply patch to get it work?

I second that. I was worried that we may need some tweaks around Arrow Java code but we don't seem to need them now. The PR now included the patch back.

File: cpp/core/memory/MemoryAllocator.cc [hunk 1]
```diff
@@ -131,11 +131,13 @@ bool StdMemoryAllocator::ReallocateAligned(void* p, uint16_t alignment, int64_t
   if (new_size <= 0) {
     return false;
   }
-  void* reallocated_p = std::realloc(p, new_size);
+  void* reallocated_p = std::malloc(new_size);
```
Comment (jinchengchenghh) [inline cpp/core/memory/MemoryAllocator.cc:134]: Why change realloc to malloc and free?

Comment (zhztheplayer) [reply cpp/core/memory/MemoryAllocator.cc:134]: This was a bug. `std::realloc` doesn't require manual copy and the `out` pointer was not updated which was wrong in previous code

**Review by jinchengchenghh — APPROVED** [review]
Change velox branch to original

**Pull request merged**